session/Session.proto

```diff
@@ -98,18 +97,28 @@ message Transaction {
     }
 
     message Iter {
-        message Req {
-            int32 id = 1;
-            bool all = 2;
-        }
         message Res {
             oneof res {
                 bool done = 1;
+                Batch.Res batch_res = 5;
+
                 Query.Iter.Res query_iter_res = 2;
                 GetAttributes.Iter.Res getAttributes_iter_res = 3;
                 Method.Iter.Res conceptMethod_iter_res = 4;
             }
         }
+        message Batch {
+            message Options {
+                int32 batch_size = 1;
+            }
+            message Req {
+                int32 id = 1;
+                Options batch_options = 2;
+            }
```

Comment:
Where are using `Iter.Batch.Req`?

Reply:
It will be used by the client's inner iterator to request another batch of results for an existing iterator.

Reply:
got it 👍